### PR TITLE
chore(web): add support for heatMap in beta

### PR DESCRIPTION
**Pull request by pyshx**
# Overview
This PR adds support for HeatMap API in web/beta's core.

### CODE REVIEW

File: web/src/beta/lib/core/engines/Cesium/Feature/HeatMap/createMeshImageData.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  const rest = pos - base;
+
+  if (sortedValues[base + 1] !== undefined) {
+    return sortedValues[base] + rest * (sortedValues[base + 1] - sortedValues[base]);
+  } else {
+    return sortedValues[base];
+  }
+}
+
+export function reverseMeshDataAndComputeValues(
```
[the remark this answers is not in the document]
Comment (pyshx) [reply web/src/beta/lib/core/engines/Cesium/Feature/HeatMap/createMeshImageData.ts]: I fixed this function, can you please check again?

https://github.com/reearth/reearth/pull/835/commits/1441a5ecc2facbddb7e4b0807989715650a33508

File: web/src/beta/lib/core/engines/Cesium/Feature/HeatMap/createMeshImageData.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+  return new Promise((resolve, reject) => {
+    const img = new Image();
+    img.onload = () => {
+      canvas.width = img.width;
+      canvas.height = img.height;
+      ctx.drawImage(img, 0, 0);
+
+      const imageData = ctx.getImageData(0, 0, img.width, img.height).data;
+      const { minValue, maxValue, outlierThreshold } = reverseMeshDataAndComputeValues(
```
Comment (keiya01) [inline web/src/beta/lib/core/engines/Cesium/Feature/HeatMap/createMeshImageData.ts:32]: I think we can skip this process when `maxValue`, `minValue` and `contourSpacing` are provided from `appearance`. Could you improve this?:pray:

Comment (pyshx) [reply web/src/beta/lib/core/engines/Cesium/Feature/HeatMap/createMeshImageData.ts]: Done.

File: web/src/beta/lib/core/engines/Cesium/Feature/index.tsx [hunk 3]
```diff
@@ -79,7 +79,7 @@ const displayConfig: Record<DataType, (keyof typeof components)[] | "auto"> = {
 // Some layer that is delegated data is not computed when layer is updated.
 // Feature's property of delegated data type is calculated when feature is loaded.
 // So in case of delegated data type, to attach property to layer, we need to use normal property before calculated.
-const PICKABLE_APPEARANCE: (keyof AppearanceTypes)[] = ["raster", "3dtiles"];
+const PICKABLE_APPEARANCE: (keyof AppearanceTypes)[] = ["raster", "3dtiles", "heatMap"];
```
Comment (pyshx) [inline web/src/beta/lib/core/engines/Cesium/Feature/index.tsx:82]: If you'll make heatMap a delegated data type, It won't support expression like
```
{
  type: "simple",
  data: {
      type: "heatMap"
  },
  heatMap: {
    valueMap: "https://dl.dropboxusercontent.com/scl/fi/cq2d4m71qg0mer2ce1b38/converted_image.png?rlkey=rhojr5xz013275duxsduivkur&dl=0",
    bounds: {
      east: 140.00000000000003,
      north: 36,
      south: 35.33333333333333,
      west: 139,
    },
    width: {
        expression: "1+2"
    }  
  }
}
```

Comment (keiya01) [reply web/src/beta/lib/core/engines/Cesium/Feature/index.tsx]: I fixed it:pray:
https://github.com/reearth/reearth/pull/835/commits/58e46946ec861e8b279fa2be5aa3cd4f36eaabde

**Pull request merged**